src/transformers/generation/utils.py

```diff
@@ -2376,8 +2403,13 @@ def generate(
         inputs_tensor, model_input_name, model_kwargs = self._prepare_model_inputs(
             inputs, generation_config.bos_token_id, model_kwargs
         )
+        # Some generation modes (e.g. assisted) need `inputs_tensor` to rerun encoder.forward()
+        if "inputs_tensor" in inspect.signature(generation_call).parameters.keys():
+            generation_mode_kwargs["inputs_tensor"] = inputs_tensor
         batch_size = inputs_tensor.shape[0]
 
+        self._validate_generation_mode(batch_size, generation_mode, generation_config, generation_mode_kwargs)
```

Comment:
This change I don't like so much -- it spreads validation code around, as opposed to being next with other validation steps at the start (which helps simplifying this long function into logical blocks)

I understand some moved checks need the batch size, I'd rather move these specific checks in the decoding method itself.